pkg/zonemgr/zone_manager.go

```diff
@@ -40,6 +40,10 @@ func (zoneMgr *ZoneManager) prepare() {
 	zoneMgr.zoneFileCache = NewZoneFileCache(nameServerIP, domain)
 
 	zoneMgr.zoneFile = NewZoneFile(zoneFileName + zoneMgr.zoneFileCache.domain)
+
+	if content, _ := zoneMgr.zoneFile.readFile(); content != nil && len(content) > 0 {
```

Comment:
please verify that error is not nil as well
if it is !=nil it is allowed to only be "not found"